File: 13/02/PageObjects/DynamicLoadingPage.cs
using OpenQA.Selenium;

namespace PageObjects 
{
    class DynamicLoadingPage : BasePage
    {
        By StartButton = By.CssSelector("#start > button");
        By FinishText = By.Id("finish");

        public DynamicLoadingPage(IWebDriver Driver) : base(Driver) { }

        public void LoadExample(int ExampleNumber)
        {
            Visit("/dynamic_loading/" + ExampleNumber);
            Click(StartButton);
        }

        public bool FinishTextPresent()
        {
            return IsDisplayed(FinishText, 10);
        }
    }
}

File: code-examples/javascript/16/03/pages/DynamicLoadingPage.js
const { set, visit, click, isDisplayed } = require("../lib/selenium-util");

const START_BUTTON = { css: "#start button" };
const FINISH_TEXT = { id: "finish" };

function load(driver) {
  set(driver);
}

async function loadExample(exampleNumber) {
  await visit("/dynamic_loading/" + exampleNumber);
  await click(START_BUTTON);
}

async function isFinishTextPresent() {
  return isDisplayed(FINISH_TEXT, 10000);
}

const DynamicLoadingPage = {
  load,
  loadExample,
  isFinishTextPresent
};

module.exports = DynamicLoadingPage;
